test/unit/AppFooterController.js

```diff
@@ -0,0 +1,37 @@
+describe('AppFooterController', function() {
+  var $controller, $scope, service, serviceFlag;
+
+  beforeEach(module('myApp.controllers'));
+
+  beforeEach(function () {
+    serviceFlag = false;
+    service = {
+      switchLayout: function(parameter){
+        serviceFlag = true;
+      }
+    }
+
+    inject(function (_$controller_, _$rootScope_) {
+        $controller = _$controller_;
+
+        $scope = _$rootScope_.$new();
+        $controller('AppFooterController', {
+            $scope: $scope,
+            LayoutSwitchService: service
+        });
+    });
+  });
+
+  // define tests
+  it('compiles', function (done) {
+    expect(true).toBe(true);
+    done();
+  });
+
+  it('calls the right function', function (done) {
+    expect(serviceFlag).toBe(false);
+    $scope.switchLayout(null);
```

Comment:
`switchLayout` sounds like a toggle. You should check the value after another call. I haven't checked the actual implementation.